### PR TITLE
create a reusable carousel component

### LINKED ISSUE #45: Creating a reusable carousel component
Many service-based websites often need to have carousels to showcase their services, work and latest activities on the home page. It would be good to have a reusable carousel(slider) component with customizable data and information. This would speed up the process of developing.
What I wish to achieve is as in this website - https://smartslider3.com/layer-slider-template/

I have been working on creating this component and will soon open a pull request for the same.

Any suggestions or features would be appreciated from the community @wdevon99 @charithccmc

### PR DESCRIPTION
**Pull request by Grumpyyash**
This PR resolves #45 
Created a reusable, customizable carousel component as described in the issue. The data including all the information, images can be passed as props to the component and it would render those. The carousel is also highly customizable and could be updated with styles and colours whatever the situation demands.

carousel

The example snippet of code on how to pass the prop is  - 

carousel-code

### CODE REVIEW

**Comment by Grumpyyash** [comment]
Could you please review this PR? @wdevon99 @charithccmc

File: src/components/FullCarouselComponent/index.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import 'react-awesome-slider/dist/styles.css';
+
+export const FullCarouselComponent = ({data}) => {
+
+  return (
+
+    <Container>
+    <div className="carousel-div">
+     <AwesomeSlider>
+     {data.map(data => (
```
Comment (subodhk01) [inline src/components/FullCarouselComponent/index.js:15]: Add a null check for if data prop in unavailable.

**Review by subodhk01 — CHANGES_REQUESTED** [review]

File: src/components/FullCarouselComponent/index.js (hunk 1 above)
Comment (Grumpyyash) [reply src/components/FullCarouselComponent/index.js:15]: Yes, I will. Thanks for reviewing it!

**Review by wdevon99 — APPROVED** [review]
Once again 'react-awesome-slider' lib is used here, we do not need wrappers for exisiting libs. We need components built by yourself :)

**Review by wdevon99 — CHANGES_REQUESTED** [review]
View previous comment, I approved by miskae

**Comment by Grumpyyash** [comment]
@wdevon99 as suggested by you, I have created the reusable component from scratch without installing any external npm package. Could you please review it now?
The data to be passed in props is - 

Code

The current carousel which can be customized using props and styles look like - 

carousel

File: src/components/CarouselComponent/index.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      </div> : null}
+      <Container>
+			{data ? <div className="carousel">
+				<Arrow direction="left" clickFunction={ previousSlide } glyph="&#9664;" />
+				<ImageSlide url={ data[currentImageIndex] } />
+				<Arrow direction="right" clickFunction={ nextSlide } glyph="&#9654;" />
+			</div> : null}
+      </Container>
+      </div>
+		);
```
Comment (wdevon99) [inline src/components/CarouselComponent/index.js:56]: Bad intendation, make sure to make the code readable and clean

**Review by wdevon99 — CHANGES_REQUESTED** [review]

**Comment by Grumpyyash** [comment]
Will improve in fresh PR :)

**Pull request closed without merging**